config/config-api/src/com/thoughtworks/go/config/Approval.java

```diff
@@ -114,21 +130,15 @@ public void validate(ValidationContext validationContext) {
                 return;
             }
             AdminsConfig adminsConfig = group.getAuthorization().getOperationConfig();
-            CruiseConfig cruiseConfig = validationContext.getCruiseConfig();
             for (Admin user : authConfig) {
-                RolesConfig roles = cruiseConfig.server().security().getRoles();
-                if (!userIsAdmin(user, cruiseConfig) && !adminsConfig.has(user, roles.memberRoles(user))) {
+                RolesConfig roles = validationContext.getServerSecurityConfig().getRoles();
+                if (!validationContext.getServerSecurityConfig().isAdmin(user) && !adminsConfig.has(user, roles.memberRoles(user))) {
```

Comment:
is `validationContext.getServerSecurityConfig()` something that needs to be pulled out of the loop?